ssas/groups.go

```diff
@@ -75,6 +129,23 @@ type GroupData struct {
 	Resources []Resource `json:"resources"`
 }
 
+// Make the GroupData struct implement the driver.Valuer interface. This method
+// simply returns the JSON-encoded representation of the struct.
+func (gd GroupData) Value() (driver.Value, error) {
+	return json.Marshal(gd)
+}
+
+// Make the GroupData struct implement the sql.Scanner interface. This method
+// simply decodes a JSON-encoded value into the struct fields.
+func (gd *GroupData) Scan(value interface{}) error {
+	b, ok := value.([]byte)
+	if !ok {
+		return errors.New("type assertion to []byte failed")
+	}
+
+	return json.Unmarshal(b, &gd)
+}
```

Comment:
Could you clarify how `Value()` and `Scan()` are being used? My understanding is that one would create these methods if a field had a custom data type. So in this case, `Group.Data` would become type `GroupData` and be parsed automatically (not calling `Scan()` or `Value()` directly). https://github.com/jinzhu/gorm/issues/47